phantazm-zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/effect/SendMessageEffect.java

```diff
@@ -1,28 +1,48 @@
 package com.github.phantazmnetwork.zombies.equipment.gun.effect;
 
-import com.github.phantazmnetwork.api.player.PlayerView;
 import com.github.phantazmnetwork.commons.AdventureConfigProcessors;
 import com.github.phantazmnetwork.commons.Namespaces;
 import com.github.phantazmnetwork.zombies.equipment.gun.GunState;
+import com.github.phantazmnetwork.zombies.equipment.gun.audience.AudienceProvider;
 import com.github.steanky.ethylene.core.ConfigElement;
 import com.github.steanky.ethylene.core.collection.ConfigNode;
 import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
 import com.github.steanky.ethylene.core.processor.ConfigProcessException;
 import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.audience.Audience;
 import net.kyori.adventure.key.Key;
 import net.kyori.adventure.key.Keyed;
 import net.kyori.adventure.text.Component;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Collection;
 import java.util.Objects;
+import java.util.function.BiConsumer;
 
+/**
+ * A {@link GunEffect} that sends a message to an {@link Audience}.
+ */
 public class SendMessageEffect implements GunEffect {
```

Comment:
question about the utility of this class: would it be used to send, for example, gold gain messages, or is it included just because

Reply:
gold gain is the message shot handler, this would be appropriate for things like notifying your purchase or that you ran out of ammo